src/main/java/com/dabsquared/gitlabjenkins/service/GitLabProjectBranchesService.java

```diff
@@ -47,23 +48,27 @@ public static class BranchLoadingException extends RuntimeException {
     }
 
     private static class BranchNamesLoader implements Function<String, List<String>> {
-        private final GitLabClient client;
+        private final GitLabApi gitLabApi;
 
-        private BranchNamesLoader(GitLabClient client) {
-            this.client = client;
+        private BranchNamesLoader(GitLabApi gitLabApi) {
+            this.gitLabApi = gitLabApi;
         }
 
         @Override
         public List<String> apply(String sourceRepository) {
             List<String> result = new ArrayList<>();
             String projectId;
             try {
-                projectId = ProjectIdUtil.retrieveProjectId(client, sourceRepository);
+                projectId = ProjectIdUtil.retrieveProjectId(gitLabApi, sourceRepository);
             } catch (ProjectIdUtil.ProjectIdResolutionException e) {
                 throw new BranchLoadingException(e);
             }
-            for (Branch branch : client.getBranches(projectId)) {
-                result.add(branch.getName());
+            try {
+                for (Branch branch : gitLabApi.getRepositoryApi().getBranches(projectId)) {
+                    result.add(branch.getName());
+                }
+            } catch (GitLabApiException e) {
+                LOGGER.log(Level.SEVERE, "failed to load branches from repository " + e.getMessage());
             }
```

Comment:
I suspect this exception was being propagated rather than logged if there was no log statement in the old code. If true, then the new code should also propagate rather than log.